flytekit/core/type_engine.py

```diff
@@ -565,9 +597,17 @@ def get_sub_type(t: Type[T]) -> Type[T]:
         """
         Return the generic Type T of the List
         """
-        if hasattr(t, "__origin__") and t.__origin__ is list:  # type: ignore
-            if hasattr(t, "__args__"):
-                return t.__args__[0]  # type: ignore
+
+        if hasattr(t, "__origin__"):
```

Comment:
The same changes need to be applied to the `DictTransformer` right?

Reply:
wasn't planning on supporting it in this pr